scripts/check_compliance.py

```diff
@@ -216,26 +231,28 @@ def run(self):
         os.environ["KCONFIG_STRICT"] = "y"
 
         try:
-            kconf = kconfiglib.Kconfig()
+            return kconfiglib.Kconfig()
         except kconfiglib.KconfigError as e:
             self.add_failure(str(e))
-            return
-
-        #
-        # Look for undefined symbols
-        #
+            raise EndTest
 
+    def check_undef_within_kconfig(self, kconf):
+        """
+        Checks that there are no references to undefined Kconfig symbols within
+        the Kconfig files
+        """
         undef_ref_warnings = [warning for warning in kconf.warnings
                               if "undefined symbol" in warning]
 
         if undef_ref_warnings:
             self.add_failure("Undefined Kconfig symbols:\n\n"
                              + "\n\n\n".join(undef_ref_warnings))
 
-        #
-        # Check for stuff being added to the top-level menu
-        #
-
+    def check_n_top_menu_items(self, kconf):
```

Comment:
I'd like to have either "max" or "too many" or "too_long_top_menu" or something like it in the function name.

Reply:
Renamed it to `check_top_menu_not_too_long()`.